lib/core/src/Cardano/Wallet.hs

```diff
@@ -518,83 +516,13 @@ restoreWallet
     -> ExceptT ErrNoSuchWallet IO ()
 restoreWallet ctx wid = do
     cp <- readWalletCheckpoint @ctx @s @t @k ctx wid
-    liftIO $ do
-        logInfo tr $ "Restoring wallet "+| wid |+"..."
-        runExceptT (networkTip nw) >>= \case
-            Left e -> do
-                logError tr $ "Failed to get network tip: " +|| e ||+ ""
-                restoreSleep @ctx @s @t @k ctx wid (currentTip cp)
-            Right tip -> do
-                restoreStep @ctx @s @t @k ctx wid (currentTip cp, tip)
+    let startTip = currentTip cp
+    let advance = restoreBlocks @ctx @s @t @k ctx wid
+    liftIO $ follow nw tr startTip advance (\b -> b ^. #header)
```

Comment:
The network layer is actually getting into shape with this. We can very much imagine the next step now with something like:

```
     liftIO $ follow nw tr startTip advance rollback (\b -> b ^. #header) 
```

Reply:
suggestion:

```hs
(\b -> b ^. #header) == view #header
```